src/cluster.c

```diff
@@ -3514,13 +3578,68 @@ clusterMsgSendBlock *clusterCreatePublishMsgBlock(robj *channel, robj *message,
     clusterMsg *hdr = &msgblock->msg;
     hdr->data.publish.msg.channel_len = htonl(channel_len);
     hdr->data.publish.msg.message_len = htonl(message_len);
-    memcpy(hdr->data.publish.msg.bulk_data,channel->ptr,sdslen(channel->ptr));
-    memcpy(hdr->data.publish.msg.bulk_data+sdslen(channel->ptr),
-        message->ptr,sdslen(message->ptr));
+    memcpy(hdr->data.publish.msg.bulk_data, channel->ptr, channel_len);
+    *dst = hdr->data.publish.msg.bulk_data + channel_len;
 
     decrRefCount(channel);
+
+    return msgblock;
+}
+
+clusterMsgSendBlock *clusterCreatePublishMsgBlock(robj *channel, robj *message, uint16_t type) {
+
+    clusterMsgSendBlock *msgblock;
+    size_t message_len;
+    unsigned char *dst;
+
+    message = getDecodedObject(message);
+    message_len = sdslen(message->ptr);
+
+    msgblock = clusterCreatePublishMsgBlockInternal(channel, message_len, type, &dst);
+    memcpy(dst, message->ptr, message_len);
+
     decrRefCount(message);
-    
+
+    return msgblock;
+}
+
+clusterMsgSendBlock *clusterCreateMPublishMsgBlock(robj *channel,
+                        robj **messages, unsigned int count, uint16_t type)
+{
+    clusterMsgSendBlock *msgblock;
+    struct {
+        robj   *obj;
+        size_t  len;
+    } *vec;
```

Comment:
Do we need to define this struct? I believe we could just loop over messages object and determine the final length of the payload.